iFixit/pulldasher
# Pulldasher: Add the 'mergeable' field

### Issue overview:

1. Pull requests with merge conflicts can end up in the **Ready** column in Pulldasher
2. QA is unable to pull up the latest changes to a pull request if there are conflicts present

![example](https://user-images.githubusercontent.com/95656772/224138987-91aa5413-0413-4ff7-92d5-dc31c46a81d8.png)

### Resolution:

* Send pull requests that have fulfilled CR/QA requirements _(but have a merge conflict)_, to the **Deploy Blocked** column.
* Update QA column sorting to place pull requests without merge conflicts above those with merge conflicts.
* Add a new Pulldasher flag to easily discern if a pull request has merge conflicts.

---

### Changes Made:

1. Added in a new field from GitHub _(mergeable)_
2. Create a new flag for the new field
3. Updated Pulldasher **QA** column sorting logic 
4. Updated Pulldasher **Ready** + **Deploy Blocked** column pull logic

>Documentation reference:
https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#get-a-pull-request

---

### CR/QA Notes:

CR: Review to https://github.com/iFixit/pulldasher/pull/356/commits/88db70109f0aa106040bd2afe84d455b2bd26df4

QA:

1. Checkout | merge with [`add-mergeable-field`](https://github.com/iFixit/pulldasher/tree/add-mergeable-field)
2. `update-pulldasher-dev --run .`
3. Navigate to https://pulldasher-dev.cominor.com/
4. Ensure that pull requests with merge conflicts:

    a.) Have a flag applied
    b.) Are in the **Deploy Blocked** column if requirements are fulfilled
    c.) Are sorted lower in the **QA** column

---

Closes #351
